contracts/tests/DualRevenueStreams.t.sol

```diff
@@ -0,0 +1,387 @@
+            emit log_named_uint("iteration", ++iterations);
+
+            _rollAndWarpToBlock(block.number + advanceBlocks);
+
+            // generate revenue for pull
+            IERC20(USDC).transfer(address(pullRevenueContract), repayment);
+        
+            // revenue for push
+            vm.prank(PUSH_REVENUE_EOA);
+            IERC20(USDC).transfer(address(spigot), repayment / 2);
+
+            // claim revenues
+            vm.startPrank(borrower);
+            emit log_string("=> claiming PUSH revenue");
+            preClaimSpigotOperatorTokens = spigot.getOperatorTokens(USDC);
+            claimed = spigot.claimRevenue(PUSH_REVENUE_EOA, USDC, bytes(""));
+            spigotOwnerTokens = spigot.getOwnerTokens(USDC);
+            spigotOperatorTokens = spigot.getOperatorTokens(USDC);
+            expectedOwnerTokens = (claimed * revenueSplit) / 100;
+            expectedOperatorTokens = preClaimSpigotOperatorTokens + (claimed * (100-revenueSplit)) / 100;
+            assertEq(spigotOwnerTokens, expectedOwnerTokens, "owner tokens dont match");
+            assertEq(expectedOperatorTokens, spigotOperatorTokens, "operator tokens dont match");
+            assertEq(spigotOperatorTokens, claimed - expectedOwnerTokens + preClaimSpigotOperatorTokens, "token totals don't add up");
+
+            emit log_string("=> claiming PULL revenue");
+            preClaimSpigotOperatorTokens = spigotOperatorTokens;
+            preClaimSpigotOwnerTokens = spigotOwnerTokens;
+            bytes memory pullClaimData = abi.encodeWithSelector(SimpleRevenueContract.claimPullPayment.selector);
+            claimed = spigot.claimRevenue(address(pullRevenueContract), USDC, pullClaimData);
+            spigotOwnerTokens = spigot.getOwnerTokens(USDC);
+            spigotOperatorTokens = spigot.getOperatorTokens(USDC);
+            spigotOperatorTokens = spigot.getOperatorTokens(USDC);
+            expectedOwnerTokens = (claimed * revenueSplit) / 100;
+            assertEq(spigotOwnerTokens, preClaimSpigotOwnerTokens + expectedOwnerTokens, "expected owner tokens don't match");
+            assertEq(spigotOperatorTokens, claimed - expectedOwnerTokens + preClaimSpigotOperatorTokens, "token totals don't match");
+            vm.stopPrank();
+
+            // check accounting
+            line.accrueInterest();
+            (, principal, interest, interestRepaid,,,,) = line.credits(line.ids(0));
+            emit log_named_uint("principal pre payment", principal);
+            emit log_named_uint("interest pre payment", interest);
+
+            // claimAndRepay ( as arbiter )
+            bytes memory repayData = abi.encodeWithSignature(
+                'trade(address,address,uint256,uint256)',
+                USDC,
+                DAI,
+                spigotOwnerTokens, // note: should be 1:1
+                spigotOwnerTokens * 10**12 // convert to DAI decimals
+            );
+            line.claimAndRepay(USDC, repayData);
+
```

Comment:
if we are actually testing repayment in this function can u also do assertions on the position data after the claimAndRepay call